datafusion-cli/src/helper.rs

```diff
@@ -197,6 +204,37 @@ pub fn unescape_input(input: &str) -> datafusion::error::Result<String> {
     Ok(result)
 }
 
+/// Splits a string which consists of multiple queries.
+pub(crate) fn split_from_semicolon(sql: String) -> Vec<String> {
+    let mut commands = Vec::new();
+    let mut current_command = String::new();
+    let mut in_single_quote = false;
+    let mut in_double_quote = false;
+
+    for c in sql.chars() {
```

Comment:
This is likely pretty brittle (likely won't handle all crazy corner cases, like for example when there are escaped quotes `\\'`, but I think it is better than what we have now. 

Thanks @berkaysynnada